xonsh/jobs.py

```diff
@@ -354,31 +356,31 @@ def fg(args, stdin=None):
         return '', 'Cannot bring nonexistent job to foreground.\n'
 
     if len(args) == 0:
-        act = tasks[0]  # take the last manipulated task by default
+        tid = tasks[0]  # take the last manipulated task by default
     elif len(args) == 1:
         try:
             if args[0] == '+':  # take the last manipulated task
-                act = tasks[0]
+                tid = tasks[0]
             elif args[0] == '-':  # take the second to last manipulated task
-                act = tasks[1]
+                tid = tasks[1]
             else:
-                act = int(args[0])
+                tid = int(args[0])
         except (ValueError, IndexError):
             return '', 'Invalid job: {}\n'.format(args[0])
 
-        if act not in builtins.__xonsh_all_jobs__:
+        if tid not in builtins.__xonsh_all_jobs__:
             return '', 'Invalid job: {}\n'.format(args[0])
     else:
         return '', 'fg expects 0 or 1 arguments, not {}\n'.format(len(args))
 
     # Put this one on top of the queue
-    tasks.remove(act)
-    tasks.appendleft(act)
+    tasks.remove(tid)
+    tasks.appendleft(tid)
 
-    job = get_task(act)
+    job = get_task(tid)
     job['bg'] = False
     job['status'] = "running"
-    print_one_job(act)
+    print_one_job(tid)
     wait_for_active_job()
```

Comment:
The alias `def fg()` may need to re-implemented, because it's running in a thread and changing controlling terminal only can be done in a main thread. But now sure, still digging.